homeassistant/components/darksky/sensor.py

```diff
@@ -390,6 +407,78 @@ def get_state(self, data):
         return state
 
 
+class DarkSkyAlertSensor(Entity):
+    """Implementation of a Dark Sky sensor."""
+
+    def __init__(self, forecast_data, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.forecast_data = forecast_data
+        self.type = sensor_type
+        self._state = None
+        self._icon = None
+        self._alerts = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(
+            self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        if self._state is not None and self._state > 0:
+            return "mdi:alert-circle"
+        return "mdi:alert-circle-outline"
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._alerts
+
+    def update(self):
+        """Get the latest data from Dark Sky and updates the states."""
+        # Call the API for new forecast data. Each sensor will re-trigger this
+        # same exact call, but that's fine. We cache results for a short period
+        # of time to prevent hitting API limits. Note that Dark Sky will
+        # charge users for too many calls in 1 day, so take care when updating.
+        self.forecast_data.update()
+        self.forecast_data.update_alerts()
+        alerts = self.forecast_data.data_alerts
+        self._state = self.get_state(alerts)
+
+    def get_state(self, data):
+        """
+        Return a new state based on the type.
+
+        If the sensor type is unknown, the current state is returned.
+        """
+        alerts = {}
+        if data is None:
+            self._alerts = alerts
+            return data
+
+        multiple_alerts = len(data) > 1
+        for i, alert in enumerate(data):
+            for attr in ALERTS_ATTRS:
+                if multiple_alerts:
+                    dkey = attr + '_' + str(i)
+                else:
+                    dkey = attr
+                alerts[dkey] = getattr(alert, attr)
+            i += 1
```

Comment:
We can remove this. `enumerate` will handle the index.

Reply:
Thanks for catching that! Fixed in https://github.com/home-assistant/home-assistant/pull/22701/commits/7ec9439d21b067e9b2ffeab922a94f08722f7570